### PR TITLE
treat the case |lambda| > 3 pi

### PR DESCRIPTION
**Pull request by Fil**
fixes https://github.com/d3/d3-geo/issues/147

### CODE REVIEW

File: src/rotation.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import compose from "./compose";
 import {asin, atan2, cos, degrees, pi, radians, sin, tau} from "./math";
 
 function rotationIdentity(lambda, phi) {
-  return [lambda > pi ? lambda - tau : lambda < -pi ? lambda + tau : lambda, phi];
+  return [lambda > pi ? (lambda + pi) % tau - pi : lambda < -pi ? (lambda - pi) % tau + pi : lambda, phi];
```
Comment (mbostock) [inline src/rotation.js]: Could this be more simply

```js
return [lambda - Math.floor((lambda + pi) / tau) * tau, phi];
```

Comment (Fil) [reply src/rotation.js]: I don't know… after 1e5 runs around the planet you start to see 1e-11 differences between those values.

I don't know which is closest to the "true" value, but more importantly, which will keep on giving the sine and cosine closest to the true values (some inconclusive tests at https://beta.observablehq.com/d/fb0735769648e8bb ).

So, whichever is fine for me at this stage. `floor` is defined in src/math.js.

**Comment by Fil** [comment]
squashed commits and clarified https://github.com/d3/d3-geo/issues/147

**Comment by mbostock** [comment]
Is there a performance impact of this change? IIRC, this code is applied to every input point when projecting spherical geometry.

**Comment by Fil** [comment]
I don't think so: the general case was two comparisons, now it is 1 Math.abs + 1 comparison.

**Pull request merged**